File: python/1829. Maximum XOR for Each Query.py
class Solution:
    def getMaximumXor(self, nums: List[int], maximumBit: int) -> List[int]:
        cum_xor = 0
        for n in nums:
            cum_xor ^= n
        max_num = (1 << maximumBit) - 1
        result = []
        n = len(nums)
        for i in range(n):
            k = max_num ^ cum_xor
            result.append(k)
            cum_xor ^= nums[-1]
            nums.pop()
        return result

File: rust/1829. Maximum XOR for Each Query.rs
impl Solution {
    pub fn get_maximum_xor(mut nums: Vec<i32>, maximum_bit: i32) -> Vec<i32> {
        let mut cum_xor = nums.iter().fold(0, |acc, &n| acc ^ n);
        let max_k = (1 << maximum_bit) - 1;

        let mut result = Vec::with_capacity(nums.len());
        for _ in 0..nums.len() {
            let k = max_k ^ cum_xor;
            result.push(k);
            cum_xor ^= nums.pop().unwrap();
        }
        result
    }
}
